### PR TITLE
Support notification for trading rewards

### PR DESCRIPTION
**Pull request by rosepuppy**
https://github.com/dydxprotocol/v4-web/assets/1062252/ac711af6-755a-4253-8ac4-17221d300afe

## Views

* New: `src/views/notifications/BlockRewardNotification/index.tsx`
  * Notification for `blockReward` based on designs
  * Includes custom background and icon

* `src/layout/NotificationsToastArea.tsx`
  * default `shouldStackNotifications` to false. This is because otherwise the reward notif gets hidden behind the fill notif. @jaredvu let me know your thoughts on this!

## Components

* `src/components/Icon.tsx`
  * Add Icon for rewardStar

## Hooks

* `src/hooks/useNotificationTypes.tsx`
  * Add handling for the `blockReward` abacus notification type

## Packages

* Update v4-localization and v4-abacus

### CODE REVIEW

File: src/views/notifications/BlockRewardNotification/index.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      notification={notification}
+      slotIcon={<Icon iconName={IconName.RewardStar} />}
+      slotTitle={stringGetter({ key: STRING_KEYS.TRADING_REWARD_RECEIVED })}
+      slotCustomContent={
+        <Styled.Details
+          items={[
+            {
+              key: 'block_reward',
+              label: stringGetter({ key: STRING_KEYS.BLOCK_REWARD }),
+              value: <Output type={OutputType.Asset} value={BLOCK_REWARD_AMOUNT} tag={TOKEN_NAME} />,
```
Comment (aforaleka) [inline src/views/notifications/BlockRewardNotification/index.tsx:43]: can we also show the green plus sign :3
image

File: src/views/notifications/BlockRewardNotification/index.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      isToast={isToast}
+      notification={notification}
+      slotIcon={<Icon iconName={IconName.RewardStar} />}
+      slotTitle={stringGetter({ key: STRING_KEYS.TRADING_REWARD_RECEIVED })}
+      slotCustomContent={
+        <Styled.Details
+          items={[
+            {
+              key: 'block_reward',
+              label: stringGetter({ key: STRING_KEYS.BLOCK_REWARD }),
```
Comment (aforaleka) [inline src/views/notifications/BlockRewardNotification/index.tsx:42]: is there a string for "Total Block Reward"?

Comment (aforaleka) [reply src/views/notifications/BlockRewardNotification/index.tsx:42]: actually ignore me, i'll update the capitalization but block reward is fine!

**Review by aforaleka — APPROVED** [review]
other than updating the stacked logic (stack if there are 3+? or stack if same type?), lgtm!

**Comment by rosepuppy** [comment]
I updated the notifications so that it would stack based on notification type:

https://github.com/dydxprotocol/v4-web/assets/1062252/92d31558-0ac4-4642-a226-189c2cbf67d9

**Pull request merged**